pkg/liqo-controller-manager/resource-request-controller/resourceRequest_controller.go

```diff
@@ -81,6 +82,32 @@ func (r *ResourceRequestReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, err
 	}
 
+	newRequireSpecUpdate := false
+	// ensure creation and deletion of the Capsule Tenant for the remote cluster
+	switch resourceReqPhase {
+	case deletingResourceRequestPhase, denyResourceRequestPhase:
+		// the local cluster does not allow the peering, ensure the permission deletion
+		if err = r.deleteClusterRoleBinding(ctx, remoteCluster); err != nil {
+			klog.Errorf("%s -> Error deleting ClusterRoleBinding: %s", remoteCluster.ClusterName, err)
+			return ctrl.Result{}, err
+		}
+		if err = r.deleteClusterRole(ctx, remoteCluster); err != nil {
+			klog.Errorf("%s -> Error deleting ClusterRole: %s", remoteCluster.ClusterName, err)
+			return ctrl.Result{}, err
+		}
+	case allowResourceRequestPhase:
+		// the local cluster allows the peering, ensure the permission
+		if err = r.ensureClusterRole(ctx, remoteCluster); err != nil {
+			klog.Errorf("%s -> Error creating ClusterRole: %s", remoteCluster.ClusterName, err)
+			return ctrl.Result{}, err
+		}
+		if err = r.ensureClusterRoleBinding(ctx, remoteCluster); err != nil {
+			klog.Errorf("%s -> Error creating ClusterRoleBinding: %s", remoteCluster.ClusterName, err)
+			return ctrl.Result{}, err
+		}
+	}
+	requireSpecUpdate = requireSpecUpdate || newRequireSpecUpdate
```

Comment:
Could also restore the tests which were already present (sorry for deleting them)